torch/distributed/_tensor/sharding_prop.py

```diff
@@ -273,12 +273,15 @@ def spec_to_strategy(spec: object) -> object:
                     needs_redistribute=needs_redistribute,
                 )
             elif isinstance(op_strategy, TupleStrategy):
-                # tuple strategy output sharding
+                # tuple strategy output sharding processing
+                # runtime selected placement strategy for each TupleStrategy input arg
+                selected_strategies: List[PlacementStrategy] = []
                 out_spec_list: List[DTensorSpec] = []
```

Comment:
nit (does not have to be in this PR): Do you think it would be clearer if we only used `selected_strategies: List[PlacementStrategy]` and refactored usages of `out_spec_list` to access the output `DTensorSpec` via the `PlacementStrategy` in `selected_strategies`?

Reply:
`out_spec_list` is more like a convenient way to get access to the list of output specs, and feed that into `OutputSharding`, otherwise we would need to loop through the `selected_strategies` again I think.. let me think more about how to make this cleaner 🤔